src/foam/core/stdlib.js

```diff
@@ -746,7 +759,24 @@ foam.LIB({
     methods: [
       function clone(o)      { return typeOf(o).clone(o); },
       function equals(a, b)  { return typeOf(a).equals(a, b); },
-      function compare(a, b) { return typeOf(a).compare(a, b); },
+      function compare(a, b) {
+        // To ensure that symmetry is present when comparing,
+        // we will always use the comparator of higher precedence.
+        var types = foam.typeOrder;
+        var typeID = function(obj) {
```

Comment:
This is going to be slow. Comparators are some of the most heavily called code in the system. Maybe add an 'ordinal' property to each of the type libs so that we can instead do:
var ta = foam.typeOf(a), tb = foam.typeOf(b);
return foam.Number.compare(ta.ordinal, tb.ordinal) || ta.compare(a, b);